### ISSUE TITLE
[TW-718] Output produced by `task completed` shouldn't use due-related colouring

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#753]
_Cory Donnelly on 2009-11-18T11:18:05Z says:_ 

 For completed tasks, colouring them according to color.overdue and color.due no longer really makes sense.

Since `task completed` is just a regular report specified within .taskrc this might be more trouble than it's worth to address -- I can't think of any other situations requiring report-specific colouring.

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2009-11-18T11:18:05Z
Modified: 2014-02-09T02:06:34Z
```

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2009-11-18T11:22:51Z says:_ 

 Good point, Cory.  If a task is completed, due date is meaningless.  Thanks.

**Comment by taskwarrior** [comment]
_David Patrick on 2009-12-01T09:32:17Z says:_ 

 new color parameter; 
color.completed=
?

down-the-road, visibility of completed items within normal reports (exclusions over-ride) 
will be toggle-able, in interactive ui, using the (Z)show-all key. 
cli example;

 $task -Z list foo

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2009-12-01T23:55:40Z says:_ 

 This is already fixed in both 1.9.0 and 1.8.5.  The colorization rules for due and overdue are suppressed if the status is completed or deleted.

Regarding the new color config (which should be a new "feature" issue), the use of "-Z" is ambiguous with the removal of the "Z" tag. This would have to be something like "--Z" instead, 
if command line options are to be supported at all.

**Comment by taskwarrior** [comment]
_David Patrick on 2009-12-02T13:10:19Z says:_ 

 status:completed tasks are currently almost never seen, as (I think all but "completed") reports filter them out, but being able to view those tasks, in mixed company with status:pending tasks, would actually be very useful in the analysis of project history, for example. If completed tasks can be viewed, especially when mixed with pending tasks, then I think users should be able to control their colors.

 the use of "-Z" is ambiguous with the removal of the "Z" tag.  
 This would have to be something like "--Z" instead, 
 if command line options are to be supported at all.

We only discussed the use of command-line options very briefly. I know that neither you nor Fredde are particularly hot on the idea, but I hope we can stay open to it a while longer.

With regards to ambiguity, I think we established that if a command-line "switch" could only be used immediately following the "task" command, 
that there could be no confusion.

One objection to the use of cli options is that a new lot of cryptic things need to be memorized. I think this is not really the case here, as we are building an interactive interface with a great lot of single-key commands. The user will have on-screen reference for these commands, along with a more detailed reference "card" available. If command line switches were ALWAYS the direct cli counterpart to the interactive key presses, then the use in one mode only re-enforces the understanding of the other. Once you are comfortable with one, the other will come naturally, and fluid use of either mode will result.

cli-options would result in;
* much stronger parallels between cli and interactive usage
* dramatically reduced key-strokes, for those that use them
* invoke behavior with two characters, that could be done otherwise, but would require more complex query construction
* nothing taken away from existing cli usage
* no new actions, just another way to perform existing ones

did I mention the key-strokes ?

**Comment by taskwarrior** [comment]
_David Patrick on 2009-12-02T13:38:42Z says:_ 

 cheekily re-opened

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2009-12-04T23:05:56Z says:_ 

 There is both a bug and a feature in this one issue.  That's bad, for tracking reasons.  once a bug is fixed, it's number goes into the ChangeLog file permanently, so there should be no revisions on closed bugs, unless they were not actually fixed, whereupon they deserve to be reopened.

The original reported by Cory has been addressed in 1.8.5.  Cory stated that tasks that had a due date, when shown in the completed report, are all colored as though they are overdue.  Because their due dates are now well into the past, even if they were completed ahead of the due date.  This has been addressed by turning off the due/overdue color rules for tasks that are deleted or completed.  I'm closing this bug again.

The second issue (feature) is that completed tasks deserve their own color setting.  Absolutely true, and should be entered as a separate feature request, as #317 is reserved for only one thing, and it is already completed and about to be released.
